src/main/java/com/rultor/agents/ec2/AmazonEC2.java

```diff
@@ -0,0 +1,139 @@
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.rultor.agents.ec2;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.services.ec2.AmazonEC2Client;
+import com.amazonaws.services.ec2.model.DescribeInstanceStatusRequest;
+import com.amazonaws.services.ec2.model.DescribeInstanceStatusResult;
+import com.amazonaws.services.ec2.model.Instance;
+import com.amazonaws.services.ec2.model.InstanceState;
+import com.amazonaws.services.ec2.model.RunInstancesRequest;
+import com.jcabi.aspects.Immutable;
+import com.jcabi.aspects.Tv;
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import javax.validation.constraints.NotNull;
+
+/**
+ * Amazon EC2 instance on demand.
+ *
+ * @author Marton Horvath (marton.horvath@m323.org)
+ * @version $Id$
+ * @since 2.0
+ */
+@Immutable
+public class AmazonEC2 implements Amazon {
+    /**
+     * AWS credentials.
+     */
+    private final transient AWSCredentials credentials;
+    /**
+     * AWS region.
+     */
+    private final transient String region;
+    /**
+     * EC2 instance type.
+     */
+    private final transient String itype;
+    /**
+     * EC2 instance key pair name.
+     */
+    private final transient String keyname;
+
+    /**
+     * Public ctor.
+     * @param cred Credentials
+     * @param zone AWS region
+     * @param type Instance type
+     * @param key Instance key pair name
+     * @checkstyle ParameterNumberCheck (12 lines)
+     */
+    public AmazonEC2(
+        @NotNull final AWSCredentials cred,
+        @NotNull final String zone,
+        @NotNull final String type,
+        @NotNull final String key) {
+        super();
+        this.credentials = cred;
+        this.region = zone;
+        this.itype = type;
+        this.keyname = key;
+    }
+
+    // @checkstyle MagicNumberCheck (30 lines)
+    @Override
+    public final Instance runOnDemand() throws IOException {
+        Instance instance = null;
+        final AmazonEC2Client client = new AmazonEC2Client(
+            this.credentials
+        );
+        client.setEndpoint(this.region);
+        final RunInstancesRequest request = this.createRequest();
+        final List<Instance> instances = client.runInstances(
+            request
+        ).getReservation().getInstances();
+        if ((instances != null) && !instances.isEmpty()) {
+            instance = instances.get(0);
+        }
+        InstanceState state = instance.getState();
+        if (state.getCode() < 16) {
+            final DescribeInstanceStatusRequest statusreq =
+                new DescribeInstanceStatusRequest();
+            statusreq.withInstanceIds(instance.getInstanceId());
+            while (state.getCode() < 16) {
+                try {
+                    TimeUnit.SECONDS.sleep(Tv.FIFTEEN);
```

Comment:
It checks the state of the instance and if it's not running (the value 16 is the code of the running instance state) we'll wait till it's state is running. The issue states that the 'runOnDemand()' method has to wait till the instance is started, so I tried to make it this way.